### ISSUE TITLE
Mutations with URIs

### ISSUE THREAD

**Issue by rubensworks** [RubenVerborgh/LDflex#7]
Currently, mutations such as `add()` and `delete()` only work with literals.
In some cases, direct URI support may also be useful, so that you could do something like `await user.friends.addUri('https://www.rubensworks.net/#me')`

I suggest to add the following mutation function variants for this:
* `.addUri()`
* `deleteUri.()`
* `setUri.()`
* `replaceUri.()`

Instead/additionally, we could also use the `Iri` or `Url` suffix.

**Comment by RubenVerborgh** [comment]
> Currently, mutations such as `add()` and `delete()` only work with literals.

I think they should work with RDF terms (as should everything else in his lib).

The question then becomes, if a string is passed instead, how do we interpret it?
Maybe in a high-level library like this, it is allowed to do `/^http?s:/ sniffing.

Alternatively/additionally, tagged template literals are great:
```JavaScript
function uri(strings) {
  return { value: strings[0], termType: 'namedNode' }
}

console.log(uri`https://example.org/`);
```

**Comment by rubensworks** [comment]
> Alternatively/additionally, tagged template literals are great:

My vote goes to this then.

`/^http?s:/` sniffing may cause too much problems I think.

So default to literals, and URIs with `uri`? Or shall we require literals to have something like a `lit` prefix as well? (Or do sniffing when no templated literal is used?)

**Comment by RubenVerborgh** [comment]
Default to terms, which indicate whether they are named nodes or literals.

When a string is entered, assume literal.

We could indeed have helpers for URIs, literals, named literals, etc. But that's only an enhancement; I'd leave that as a separate issue.

**Comment by rubensworks** [comment]
I guess we should first look into resolving #3 then, which is blocked by comunica/comunica#358.
